src/main/java/net/dries007/tfc/common/blockentities/rotation/WindmillBlockEntity.java

```diff
@@ -220,4 +227,31 @@ public Node getRotationNode()
     {
         return node;
     }
+
+    public boolean hasFullIdenticalSet()
```

Comment:
Two changes:

- This method should stay, but just be an accessor for the underlying variable - the public accessor shouldn't do any computation. Right now it's not clear that `windmill.hasFullIdenticalSet()` is computation, but `windmill.hasFulIdenticalSet` is cached (also why the variable should be made private)
- The _logic_ present here should be moved to a private method, e.g. `checkForFullIdenticalSet()`, which updates the value of this variable. This method needs to be called from two places:
  - `loadAdditional` - when we load the windmill from nbt
  - `setAndUpdateSlots` - when the inventory is modified

This way we only recompute it when the inventory changes - not every client tick, and it's clear what the accessor is for.